src/utils/rateLimiter.util.ts

```diff
@@ -0,0 +1,27 @@
+import { RedisService } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class RateLimiterService {
+  private readonly logger = new Logger(RateLimiterService.name);
+
+  constructor(private readonly redisService: RedisService) {}
+
+  async isRateLimited(senderId: string, repoId: string): Promise<boolean> {
+    const client = this.redisService.getClient();
+    const key = `rate_limit:${senderId}:${repoId}`;
+    const currentCount = await client.get(key);
+
+    if (currentCount && parseInt(currentCount) >= 5) {
```

Comment:
PullProofAI

Consider adding error handling for the case where `parseInt` fails or throws an exception.
```diff
+     const currentCount = await client.get(key);
+
+     if (!currentCount) {
+         throw new Error('Failed to retrieve current count from Redis');
+     }
+
+     let parsedCount;
+     try {
+         parsedCount = parseInt(currentCount);
+     } catch (error) {
+         throw new Error('Failed to parse current count as integer');
+     }
+
+     if (isNaN(parsedCount)) {
+         throw new Error('Parsed count is not a number');
+     }
+
+     if (parsedCount >= 5) {
```



<!-- This is an auto-generated comment by PullProofAI -->